UNIFY-412 update example recipe readme and examples

jaffrepaul opened the pull request:
# DO NOT MERGE

This PR **will not be merged** but its content will be folded into an upcoming release. This contains many (but not all) of the updates needed to document the `cypress.json` -> `cypress.config.js` changes.

https://cypress-io.atlassian.net/browse/UNIFY-412

**Note:** batch lint files after review & before merge

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cypress-io/cypress-example-recipes?pullRequest=743) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cypress-io/cypress-example-recipes?pullRequest=743) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cypress-io/cypress-example-recipes?pullRequest=743) it.</sub>

lmiller1990 commented on README.md:
```diff
@@ -131,8 +131,8 @@ Recipe | Description
 [Visiting 2nd domain](./examples/server-communication__visit-2nd-domain) | Visiting two different domains from two different tests and passing value from one test to another
-[Pass value between specs](./examples/server-communication__pass-value-between-specs) | Pass a value from spec to spec via the plugin file using [cy.task](https://on.cypress.io/task)
-[Stream test results](./examples/server-communication__stream-tests) | Streams each test result from the browser to the plugins to an external process via IPC
+[Pass value between specs](./examples/server-communication__pass-value-between-specs) | Pass a value from spec to spec via the setupNodeEvents function in the Cypress configuration using [cy.task](https://on.cypress.io/task)
```
Might be good to make `setupNodeEvents` have backticks to emphasize it is a specific function that is part of the configuration.

jaffrepaul replied:
good catch, I'll update these.

lmiller1990 reviewed (commented):
Are all these changes intentional? It looks like some formatting has been applied to everything - not sure if this was intentional or not, but makes it kind of hard to review.

Probably okay for now but would be good to enforce some automated linting across our examples repo.

cowboy commented:
> Are all these changes intentional? It looks like some formatting has been applied to everything - not sure if this was intentional or not, but makes it kind of hard to review.

Every `cypress.json` + plugins file had to be merged and reformatted before we started going through them, are you talking about those files, or something else?

cowboy commented:
> Every `cypress.json` + plugins file had to be merged and reformatted before we started going through them, are you talking about those files, or something else?

I see what you're saying now. @jaffrepaul what's the LOE to split this into 2 PRs, one that has just the formatting changes (quotes, whitespace), and another from that PR that has all the other changes?

jaffrepaul commented:
@cowboy @lmiller1990 No problem on the linting. I ran the command at the end to get everything consistent. I'll split that off and we can handle separately.

jaffrepaul commented:
OK @lmiller1990 @cowboy I reverted the linting & added `setUpNodeEvents` for clarity. I kept them as two commits for clarity. 

Note: 
- most of the linting was batched via the eslint command but some was picked up via the Husky pre commit hook while I was editing some files by hand. I reverted these too, which in isolation felt a little silly. But there's far less noise in this PR now and I can just scoop all these up with the command separately. There are a 1-2 files that are tracking the style revert as a result of being included with other stuff. Just know that will all be rectified later. 

- in the interest of time I wrapped `setUpNodeEvents` in .md files and comments as well. I can edit that if it's weird
